Add nengo_spinnaker_setup command.

mossblaser opened the pull request:
This command automatically creates nengo_spinnaker config files (in the right places) based on easily-answered questions.

This PR is waiting on:
- [x] project-rig/rig#165 being merged and released
- [x] The setup.py dependency for rig being updated accordingly
- [x] Update README
- [x] Remove system-wide config option.

An example invocation:

```
$ nengo_spinnaker_setup
:0: FutureWarning: IPython widgets are experimental and may change in the future.
What type of SpiNNaker system to you have?
    0: A single four-chip 'SpiNN-3' board
    1: A single forty-eight-chip 'SpiNN-5' board
    2: Multiple forty-eight-chip 'SpiNN-5' boards
    3: Other
Select an option 0-3: 0

Would you like to auto-detect the SpiNNaker system's IP address?
    0: Auto-detect
    1: Manually Enter IP address or hostname
Select an option 0-1 (default: 0): 0

Make sure the SpiNNaker system is switched on and is not booted.
<Press enter to continue>

Discovering attached SpiNNaker systems...

Successfully created config file in /home/jonathan/.config/nengo/nengo_spinnaker.conf
```

Note that if a config file already exists, `--force` must be used:

```
$ nengo_spinnaker_setup
Config file /home/jonathan/.config/nengo/nengo_spinnaker.conf already exists. Use --force to overwrite.
$ nengo_spinnaker_setup --force
[...]
```

Both user-specific and project-specific config files can be generated:

```
nengo_spinnaker_setup --user
nengo_spinnaker_setup --project
```

mossblaser committed:
Add nengo_spinnaker_setup command.

This command automatically creates nengo_spinnaker config files (in the right
places) based on easily-answered questions.

Add Rig v0.5.3 dependency.

This dependency ensures the features required by nengo_spinnaker_setup are
present.

mossblaser commented:
(Note: the failed travis run is due to the unreleased state of project-rig/rig#165 -- travis should be re-run   when this has been released).

mundya commented on nengo_spinnaker/scripts/nengo_spinnaker_setup.py:
```diff
@@ -0,0 +1,83 @@
+                            help="Create a system-wide config file.")
+    file_group.add_argument("--user", "-u", dest="file",
+                            const="user", action="store_const",
+                            help="Create a user user-specific config file "
```
"user user-specific"

mundya commented on nengo_spinnaker/scripts/nengo_spinnaker_setup.py:
```diff
@@ -0,0 +1,83 @@
+    file_group = parser.add_mutually_exclusive_group()
+    file_group.add_argument("--system", "-s", dest="file",
+                            const="system", action="store_const",
+                            help="Create a system-wide config file.")
```
Actually not sure this will work without other changes to nengo_spinnaker.

mossblaser replied:
Ah yes... I should have grocked https://github.com/project-rig/nengo_spinnaker/blob/34910e6851894445cd660693411dd49ee053baf6/nengo_spinnaker/utils/paths.py#L12 before using it. Shall I remove this for now?

mundya replied:
Yeah, keep removing it as a separate commit though so it can be reverted later down the line. Thanks!

mundya commented:
Cool :)

mossblaser committed:
Remove --system option from nengo_spinnaker_setup

Removed until a future time where an *editable* system wide config file exists
as suggested by @mundya.

mossblaser commented:
OK this should be good to go! Feel free to squash and merge when you're happy.
